### ISSUE TITLE
Random crash

### ISSUE THREAD

**Issue by Latka808** [RoystonS/BetterBravoLights#23]
**Software version**

0.6.0.0

**Describe the bug**

I had just flown from PHNG to PHMK and landed. Taxi'd in, and refueled. Started the plane back up, and was talking to ATC for taxi clearance, and had a crash to desktop of MSFS and BetterBravoLights. Please understand that I am not mad, I'm just curious about what happened. Not sure if it was an MSFS thing, a BetterBravoLights thing, or something unrelated. The high number of CTDs and lack of any error logging in MSFS is frustrating. :\

**Other software**

- MSFS: Steam

**Additional context**
I ran "View Reliability History" in Windows and it showed me that both FlightSimulator.exe and BetterBravoLights.exe had "Stopped Working." (Control Panel\All Control Panel Items\Security and Maintenance\Problem Details)

Here's the information that it showed me from FlightSimulator.exe:
Source
FlightSimulator.exe

Summary
Stopped working

Date
‎11/‎30/‎2021 19:48

Status
Report sent

Description
Faulting Application Path:	C:\Program Files (x86)\Steam\steamapps\common\MicrosoftFlightSimulator\FlightSimulator.exe

Problem signature
Problem Event Name:	APPCRASH
Application Name:	FlightSimulator.exe
Application Version:	1.21.13.0
Application Timestamp:	00000000
Fault Module Name:	FlightSimulator.exe
Fault Module Version:	1.21.13.0
Fault Module Timestamp:	00000000
Exception Code:	c0000005
Exception Offset:	0000000002483577
OS Version:	10.0.19044.2.0.0.256.48
Locale ID:	1033
Additional Information 1:	54d2
Additional Information 2:	54d2a9747ca914384141a26ff29a12eb
Additional Information 3:	a770
Additional Information 4:	a7706ce3baf24deea7469f253ee5f9b5

Extra information about the problem
Bucket ID:	667fa6f8d3909a703f717e361a01eb07 (2265730858373737223)
----------------------------------------------

Here's the info from BetterBravoLights:
Source
BetterBravoLights

Summary
Stopped working

Date
‎11/‎30/‎2021 19:49

Status
Report sent

Description
Faulting Application Path:	C:\Program Files (x86)\BetterBravoLights_gZBm4\Program\BetterBravoLights.exe

Problem signature
Problem Event Name:	APPCRASH
Application Name:	BetterBravoLights.exe
Application Version:	0.6.0.0
Application Timestamp:	614cbf2c
Fault Module Name:	KERNELBASE.dll
Fault Module Version:	10.0.19041.1387
Fault Module Timestamp:	0b9a844a
Exception Code:	e0434352
Exception Offset:	0000000000034f69
OS Version:	10.0.19044.2.0.0.256.48
Locale ID:	1033
Additional Information 1:	319d
Additional Information 2:	319df3396735776e06e788ea83f355bb
Additional Information 3:	ff92
Additional Information 4:	ff92b0239c28fa9cf388aff689e28fb8

Extra information about the problem
Bucket ID:	3abf95950e657e16ebf20a1edf8cd1b2 (2013683111103877554)
----------------------------------------------------

Please let me know if there are any other logs or diagnostic stuff I can get for you. Thanks! -Andy

**Comment by RoystonS** [comment]
Hi @Latka808 

Yeah, this will be tricky. As you say, MSFS will frequently CTD all by itself, with no logging output.  BetterBravoLights is obviously communicating with MSFS, so if MSFS crashes it _can_ take out BBL too. (MSFS went at 19:48, BBL at 19:49.)

But I can't say with _any_ certainty that MSFS crashed by itself, or that BBL had no part in the crash... but it's _likely_ that MSFS crashed by itself and that BBL was an innocent bystander.

I'm definitely interested that BBL was reported as 'stopped working'. As you can imagine, I do have specific code in there to handle errors from MSFS, and to cope when MSFS CTDs and attempt to shut down BBL automatically in the same way as if MSFS had exited cleanly.  It could be that something wasn't quite right with that 'exit on MSFS crash' code.

Let's keep an eye on it. If you're getting regular CTDs we should then try the exact same setup (same community folder entries and other apps etc) but without BBL to see if that changes the frequency of the CTDs. Given the dreadful monitoring and reporting from MSFS that's the best I think we can do right now?  I'll take a look at the exit-on-crash code to see if there's anything obvious there.

**Comment by Latka808** [comment]
Hi Royston, I was figuring that might be along the lines of what you replied. It’s not your fault – they need to give everyone better tools to figure out what the hell went wrong. Will keep an eye on things for now. Just wanted to bring it up in case other folks were experiencing similar things. -Andy From: Royston ShufflebothamSent: Wednesday, December 1, 2021 00:23To: RoystonS/BetterBravoLightsCc: Latka808; MentionSubject: Re: [RoystonS/BetterBravoLights] Random crash (Issue #23) Hi @Latka808Yeah, this will be tricky. As you say, MSFS will frequently CTD all by itself, with no logging output. BetterBravoLights is obviously communicating with MSFS, so if MSFS crashes it can take out BBL too. (MSFS went at 19:48, BBL at 19:49.)But I can't say with any certainty that MSFS crashed by itself, or that BBL had no part in the crash... but it's likely that MSFS crashed by itself and that BBL was an innocent bystander.I'm definitely interested that BBL was reported as 'stopped working'. As you can imagine, I do have specific code in there to handle errors from MSFS, and to cope when MSFS CTDs and attempt to shut down BBL automatically in the same way as if MSFS had exited cleanly. It could be that something wasn't quite right with that 'exit on MSFS crash' code.Let's keep an eye on it. If you're getting regular CTDs we should then try the exact same setup (same community folder entries and other apps etc) but without BBL to see if that changes the frequency of the CTDs. Given the dreadful monitoring and reporting from MSFS that's the best I think we can do right now? I'll take a look at the exit-on-crash code to see if there's anything obvious there.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by RoystonS** [comment]
Hiya. Whilst working on the new 'variable list' feature in BBL I did find an issue with the subscription and unsubscription of variables. Most of the code on the internet that says that recommends a particular way of unsubscribing from variables is actually wrong, but it's only really obvious when you have subscribed to, and unsubscribed from, 1000 variables. I don't think you'll have been hitting that (unless you have some _super_ complex configurations!), but I've fixed that up anyway in v0.7.0.

https://github.com/RoystonS/BetterBravoLights/commit/42d872585f37ac93474b218f2832cb5464e71cc2
